.github/workflows/trivy.yaml

```diff
@@ -0,0 +1,49 @@
+name: Trivy
+
+on:
+  push:
+    branches:
+      - "main"
+  schedule:
+    - cron: '0 20 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  image-scan:
+    permissions:
+      contents: read  # Required for actions/checkout to fetch code
+    name: Image Scan
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+
+      - name: Build Controller Docker image
+        run: |
+          make docker-build.controller TAG=${{ github.sha }}
+
+      - name: Build Ext-Proc Docker image
+        run: |
+          make docker-build.extproc TAG=${{ github.sha }}
+
+      - name: Run Trivy vulnerability scanner for Controller
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ghcr.io/envoyproxy/ai-gateway/controller:${{ github.sha }}
+          format: 'table'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW'
+          vuln-type: 'os,library'
+          exit-code: '1'  # Fail workflow on detected vulnerabilities
+          ignore-unfixed: true  # Ignore unfixed vulnerabilities
+
+      - name: Run Trivy vulnerability scanner for Ext-Proc
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ghcr.io/envoyproxy/ai-gateway/extproc:${{ github.sha }}
+          format: 'table'
+          severity: 'CRITICAL,HIGH,MEDIUM,LOW'
+          vuln-type: 'os,library'
+          exit-code: '1'  # Fail workflow on detected vulnerabilities
+          ignore-unfixed: true  # Ignore unfixed vulnerabilities
```

Comment:
maybe adding this part of the existing docker build flow? like adding this after https://github.com/envoyproxy/ai-gateway/blob/cb6b2e0f12fb6bc2e23941eff6039cf78d5693ca/.github/workflows/docker_builds_template.yaml#L47 ?

Reply:
is that costly? (i have never used trivy before)

Reply:
trivy is commonly use in os projects for container security and vulnerability scanning.

adding to docker_builds_template.yaml fine, but the problem is it runs on every pull request/commit and it will block all new prs if new vulnerability detect in existing components. so i would suggest to keep it as separate wf.
scheduling is commonly used when cost factor is considered so if needed we can schedule it per week rather that per day.